.github/workflows/scripts/pr_workflow.py

```diff
@@ -282,12 +288,14 @@ def issue_comment_label_actions(
                 LOGGER.info(f"Processing label: {label}, action: {action}")
 
                 if action[CANCEL_ACTION] or self.event_action == "deleted":
+                    self.dismiss_pr_approval()
                     if label_in_pr:
                         LOGGER.info(f"Removing label {label}")
                         self.pr.remove_from_labels(label=label)
 
                 elif not label_in_pr:
                     self.add_pr_label(label=label)
+                    self.approve_pr()
```

Comment:
wouldn't this approave the pr for all user labels?
pr should be approved only if self.user_login != pr owner

Reply:
>wouldn't this approave the pr for all user labels?
shoot, yeah, I was sure it was indented under the first `if` block, will fix tomorrow

>pr should be approved only if self.user_login != pr owner
let's discuss about this, I think there are more scenarios to consider

Reply:
Checking for team membership now. I've done a workaround until we can get the bot into the org but the logic is there.